src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1479,7 +1479,7 @@ internal sealed class SourceComplexParameterSymbol : SourceComplexParameterSymbo
             SyntaxReference syntaxRef,
             bool isParams,
             bool isExtensionMethodThis,
-            DeclarationScope scope)
+            DeclarationScope? scope)
```

Comment:
>DeclarationScope? scope [](http://example.com/codeflow?start=12&length=23)

Is this change needed? Can we get here from a `LambdaParameterSymbol`?

Reply:
`WithCustomModifiersAndParamsCore` copies `this.DeclaredSymbol` (which is `Nullable<DeclarationScope>`) using this constructor.
It similarly calls the `SourceComplexParameterSymbolWithCustomModifiersPrecedingRef` constructor.

Reply:
Is `WithCustomModifiersAndParamsCore()` used for `LambdaParameterSymbol`? Otherwise, we'd presumably need to copy `EffectiveScope` as well.

Reply:
`WithCustomModifiersAndParamsCore` isn't used for lambdas. It's used for OHI scenarios.